### PR TITLE
Improve processHtml #9413

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Review by vbradnitski — APPROVED** [review]

File: modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/HtmlProcessorParams.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import com.enonic.xp.annotation.PublicApi;
+import com.enonic.xp.portal.html.HtmlDocument;
+import com.enonic.xp.portal.html.HtmlElement;
+
+@PublicApi
+public class HtmlProcessorParams
```
Comment (rymsha) [inline modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/HtmlProcessorParams.java:11]:
```suggestion
public final class HtmlProcessorParams
```

File: modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/HtmlProcessorParams.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            defaultElementProcessor.accept( element, elementProcessor );
+        }
+    }
+
+    public static Builder create()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
```
Comment (rymsha) [inline modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/HtmlProcessorParams.java:68]:
```suggestion
    public static final class Builder
```

File: modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/HtmlProcessorParams.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        private HtmlDocument htmlDocument;
+
```
Comment (rymsha) [inline modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/HtmlProcessorParams.java:71]: Add private constuctor

File: modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/ProcessHtmlParams.java [hunk 4]
```diff
@@ -20,6 +21,10 @@
 
     private String imageSizes;
 
+    private Function<HtmlProcessorParams, String> customHtmlProcessor;
+
+    private boolean useCustomMacrosProcessing = false;
```
Comment (rymsha) [inline modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/ProcessHtmlParams.java:26]: It should be called `processMacros` with default `true`

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

**Pull request merged**